src/t_zset.c

```diff
@@ -2952,7 +2952,12 @@ static void zrangeResultFinalizeClient(zrange_result_handler *handler,
 static void zrangeResultBeginStore(zrange_result_handler *handler, long length)
 {
     UNUSED(length);
-    handler->dstobj = createZsetListpackObject();
+    if (server.zset_max_listpack_entries > 0) {
+        handler->dstobj = createZsetListpackObject();
+    } else {
+        /* Use skiplist if listpack is disabled to avoid redundant conversion on add. */
+        handler->dstobj = createZsetObject();
+    }
```

Comment:
i don't like to repeat this condition everywhere... maybe that signifies that we should have a wrapper method that implements that logic.

but more importantly, i don't like the fact we we fixed zsetConvert (to be on the safe side), but this fix is unreachable / untested, because of these checks on creation.

I see you said you prefer to keep it since it's an optimization, but i think this optimization is unnecessary, it's not a common case / configuration, and the impact is probably low anyway.
converting an empty listpack will be very fast, and the only overhead is the malloc / free.
so i'd rather cleaner code in that case..

p.s. if we do want to optimize the creation, we should be looking at the `length` argument, so when we know the size of the result ahead of time, we can create a skiplist object in advance, instead of converting it when it reaches the threshold (that conversion does have a price)

Reply:
Okay, I'll remove those checks, I don't think that will have any noticable impact

Reply:
FYI: added that other optimization idea here: https://github.com/redis/redis/pull/10789 didn't prove it's any good yet, but didn't wanna forget it..